docs/libraries/azure-core/reference/linter.md

```diff
@@ -59,5 +59,6 @@ Available ruleSets:
 | `@azure-tools/typespec-azure-core/response-schema-problem`                                                           | Warn about operations having multiple non-error response schemas.                                                                                    |
 | `@azure-tools/typespec-azure-core/rpc-operation-request-body`                                                        | Warning for RPC body problems.                                                                                                                       |
 | `@azure-tools/typespec-azure-core/spread-discriminated-model`                                                        | Check a model with a discriminator has not been used in composition.                                                                                 |
+| `@azure-tools/typespec-azure-core/use-standard-integer`                                                              | Use only types that map to int32 or int64.                                                                                                           |
```

Comment:
I think this is actively harmful. It will push service teams to be less precise than they can/should be (e.g. the service exposes an 8 bit unsigned integer for an embedding vector - should we now say that it is an int32? And do that at the expense of DPG emitters that actually *do* support unsigned byte sized types?). It is especially bad since reviewers have no reliable way to understand what the intent of the service spec author was. And, nitpick, we also tell spec authors to not use int64 because ECMA script - that would be `safeint` in TypeSpec.

If we insist on this rule, we need to better understand what tools it protects. E.g. if we are not generating code from the generated openapi/swagger, then it doesn't matter what auto rest does. If we *do* have to protect downstream tools, if is probably better that the autorest-openapi emitter emits the formats.

Reply:
This has been replaced with a rule that simply checks for generic numeric types (numeric, decimal, float, integer). @johanste  please take a look at the updated logic and let me know your thoughts.